src/main/g8/$name__norm$-impl/src/main/scala/$package$/impl/$name__Camel$Aggregate.scala

```diff
@@ -0,0 +1,268 @@
+    * loaded, you don't need to replay all the events, just the ones since the
+    * snapshot. Hence, a JSON format needs to be declared so that it can be
+    * serialized and deserialized when storing to and from the database.
+    */
+  implicit val format: Format[$name;format="Camel"$State] = Json.format
+}
+
+/**
+  * This interface defines all the events that the $name;format="Camel"$Aggregate supports.
+  */
+sealed trait $name;format="Camel"$Event extends AggregateEvent[$name;format="Camel"$Event] {
+  def aggregateTag: AggregateEventTag[$name;format="Camel"$Event] = $name;format="Camel"$Event.Tag
+}
+
+object $name;format="Camel"$Event {
+  val Tag: AggregateEventTag[$name;format="Camel"$Event] = AggregateEventTag[$name;format="Camel"$Event]
+}
+
+/**
+  * An event that represents a change in greeting message.
+  */
+case class GreetingMessageChanged(message: String) extends $name;format="Camel"$Event
+
+object GreetingMessageChanged {
+
+  /**
+    * Format for the greeting message changed event.
+    *
+    * Events get stored and loaded from the database, hence a JSON format
+    * needs to be declared so that they can be serialized and deserialized.
+    */
+  implicit val format: Format[GreetingMessageChanged] = Json.format
+}
+
+/**
+  * This is a marker trait for commands.
+  * We will serialize them using Akka's Jackson support that is able to deal with the replyTo field.
+  * (see application.conf)
+  */
+trait $name;format="Camel"$CommandSerializable
+
+/**
+  * This interface defines all the commands that the $name;format="Camel"$Aggregate supports.
+  */
+sealed trait $name;format="Camel"$Command
+    extends $name;format="Camel"$CommandSerializable
+
+/**
+  * A command to switch the greeting message.
+  *
+  * It has a reply type of [[Confirmation]], which is sent back to the caller
+  * when all the events emitted by this command are successfully persisted.
+  */
+case class UseGreetingMessage(message: String, replyTo: ActorRef[Confirmation])
+    extends $name;format="Camel"$Command
+
+/**
+  * A command to say hello to someone using the current greeting message.
+  *
+  * The reply type is String, and will contain the message to say to that
+  * person.
+  */
+case class Hello(name: String, replyTo: ActorRef[Greeting])
+    extends $name;format="Camel"$Command
+
+sealed trait $name;format="Camel"$Reply
+
+object $name;format="Camel"$Reply {
+  implicit val format: Format[$name;format="Camel"$Reply] =
+    new Format[$name;format="Camel"$Reply] {
+
+      override def reads(json: JsValue): JsResult[$name;format="Camel"$Reply] = {
+        if ((json \ "state").isDefined)
+          Json.fromJson[Greeting](json)
+        else
+          Json.fromJson[Confirmation](json)
+      }
+
+      override def writes(o: $name;format="Camel"$Reply): JsValue = {
+        o match {
+          case conf: Confirmation => Json.toJson(conf)
+          case state: Greeting    => Json.toJson(state)
+        }
+      }
+    }
+}
```

Comment:
@renatocaval I don't think the `sealed trait $name;format="Camel"$Reply` supertype is necessary anymore, right?

Reply:
I dropped the `XyzReply` supertype. It is no longer necessary.